### PR TITLE
Added Comment section

### PR DESCRIPTION
**Pull request by saumy-sh**
Issue: #212 
I have added comment section below each recipe, so users can comment and their comments will be saved in comments database with recipe id, username(who commented),content and date of comment. I have attached video as well.

https://github.com/user-attachments/assets/785eabcc-e0af-47fa-a9eb-a9053637ff9b

### CODE REVIEW

**Comment by AlfiyaSiddique** [comment]
@saumy-sh can you please  add a video of output as the backend is not deployed for preview hence it is not working on preview.

**Comment by saumy-sh** [comment]
I have also updated pr to add video as well.

File: backend/server.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ app.use(express.urlencoded({ limit: "10mb" }));
 
 // CORS configuration
 const allowedOrigins = [
+    "http://localhost:5173",
```
Comment (AlfiyaSiddique) [inline backend/server.js:20]: remove this

File: backend/server.js [hunk 2]
```diff
@@ -72,5 +73,6 @@ try {
             });
         });
 } catch (error) {
+    // console.log(process.env.DATABASE);
     console.log(error);
```
Comment (AlfiyaSiddique) [inline backend/server.js:77]: remove this console.log please do not print any env information

File: frontend/src/Pages/Signup.jsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { faEye, faEyeSlash } from "@fortawesome/free-solid-svg-icons";
 import validate from "../../common/validation.js";
 import axios from "axios";
 import { toast } from "react-toastify";
-import useGoogleAuth from "../../common/useGoogleAuth"
+
```
Comment (AlfiyaSiddique) [inline frontend/src/Pages/Signup.jsx:10]: Same comment here as well.
All code related to Google signing is removed can you please fix it

**Comment by AlfiyaSiddique** [comment]
@saumy-sh some comments above please fix that

File: frontend/src/Pages/Login.jsx [hunk 4]
```diff
@@ -6,7 +6,7 @@ import axios from "axios";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faEye, faEyeSlash } from "@fortawesome/free-solid-svg-icons";
 import { useNavigate } from "react-router-dom";
-import useGoogleAuth from "../../common/useGoogleAuth"
+
```
[the remark this answers is not in the document]
Comment (saumy-sh) [reply frontend/src/Pages/Login.jsx:9]: yeah sure in my local machine that was not running that's why i removed it temporarily. Should i add it back?

File: frontend/src/Pages/Signup.jsx (hunk 3 above)
Comment (saumy-sh) [reply frontend/src/Pages/Signup.jsx:10]: Sure!!

**Comment by saumy-sh** [comment]
I have added the google auth changes can you check it once?

File: .gitignore [hunk 5]
```diff
@@ -22,5 +22,8 @@ dist-ssr
 *.njsproj
 *.sln
 *.sw?
-
+package-lock.json
+package.json
+node_modules
+*.env
 /*/.env
```
Comment (AlfiyaSiddique) [inline .gitignore:29]: @saumy-sh other than node_modules please remove all all three additions. Because of this preview deployment is failing

**Comment by AlfiyaSiddique** [comment]
@saumy-sh deployment preview is failing, some comments on that, this is the error that we got from the logs:
image
Please fix it

File: .gitignore (hunk 5 above)
Comment (saumy-sh) [reply .gitignore:29]: I will resolve this.Any other changes that I need to do?

**Comment by AlfiyaSiddique** [comment]
@saumy-sh you have deleted the file useGoogleAuth.js from the common folder. That file is important. 
This also results in deployment errors. 
>: RollupError: Could not resolve "../../common/useGoogleAuth" from "src/Pages/Login.jsx"
9:08:19 PM:     at error 

Please do not delete any file if it's not working locally. You could comment it out and then remove the comments again while pushing code.
Now the PR can only be merged after these problems are resolved.
Please fix it.

**Comment by saumy-sh** [comment]
I have done the necessary change.

**Comment by saumy-sh** [comment]
I had closed earlier pull request that I had opened, now this is the pull request that I am adding to the mentioned issue. I hope there isn't any issue with this?

**Review by AlfiyaSiddique — APPROVED** [review]

**Comment by AlfiyaSiddique** [comment]
@saumy-sh, there is no issue with opening a new PR, but it's a request to manage the changes of an issue in one PR only unless it is too necessary to open a new PR. This will teach you how to manage conflicts, commits, and reverts. That's the reason.
Thanks for your contribution,

**Comment by saumy-sh** [comment]
Thanks for merging it and being so patient.

**Comment by AlfiyaSiddique** [comment]
@saumy-sh just want to confirm that you did not add the comment storage in database functionality with this PR, Am I right?

**Comment by AlfiyaSiddique** [comment]
As I log in from another account I can't see the comment I did from a different account on the same recipe.
Even I can't see the comments once I logged out and login again

**Comment by saumy-sh** [comment]
No I added the database for comments, you can see that in models folder. You will have to logout and sign in again to see the changes as I have created a localStorage token for login

**Comment by saumy-sh** [comment]
Screenshot 2024-10-18 000457

I can see the comments here

**Comment by saumy-sh** [comment]
Can you check if the comment collection is created in database?

**Comment by AlfiyaSiddique** [comment]
I can't see it in deployment, is the above ss from the deployed app?
image

And yes the records are created in the database.
No worries. It looks like there is some bug. Someone's working on it. No worries.
If you can think of any config settings needed in deployment for this feature that I forgot to add that might cause this bug then please let me know.
Thanks for the info.

**Comment by saumy-sh** [comment]
Ok sure! I don't think any other backend changes are required, just you have to give your mongodb uri in .env folder

**Comment by saumy-sh** [comment]
> I can't see it in deployment, is the above ss from the deployed app?
Yes it's from the deployed link

**Pull request merged**